task/servicetest/servicetest.go

```diff
@@ -1039,6 +984,214 @@ func testManualRun(t *testing.T, s *System) {
+		t.Fatalf("wrong task ID on created task run: got %s, want %s", rc1.Created.TaskID, task.ID)
+	}
+
+	// Update the run state to Started; normally the scheduler would do this.
+	if err := sys.TaskControlService.UpdateRunState(sys.Ctx, task.ID, rc1.Created.RunID, startedAt, backend.RunStarted); err != nil {
+		t.Fatal(err)
+	}
+
+	// Mark the second run finished.
+	if err := sys.TaskControlService.UpdateRunState(sys.Ctx, task.ID, rc1.Created.RunID, startedAt.Add(time.Second), backend.RunSuccess); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := sys.TaskControlService.FinishRun(sys.Ctx, task.ID, rc1.Created.RunID); err != nil {
+		t.Fatal(err)
+	}
+
+	// Limit 1 should only return the earlier run.
+	runs, _, err := sys.TaskService.FindRuns(sys.Ctx, influxdb.RunFilter{Task: task.ID, Limit: 1})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(runs) != 1 {
+		t.Fatalf("expected 1 run, got %v", runs)
+	}
+	if runs[0].ID != rc0.Created.RunID {
+		t.Fatalf("retrieved wrong run ID; want %s, got %s", rc0.Created.RunID, runs[0].ID)
+	}
+	if exp := startedAt.Format(time.RFC3339Nano); runs[0].StartedAt != exp {
+		t.Fatalf("unexpectedStartedAt; want %s, got %s", exp, runs[0].StartedAt)
+	}
+	if runs[0].Status != backend.RunStarted.String() {
+		t.Fatalf("unexpected run status; want %s, got %s", backend.RunStarted.String(), runs[0].Status)
+	}
+	if runs[0].FinishedAt != "" {
+		t.Fatalf("expected empty FinishedAt, got %q", runs[0].FinishedAt)
+	}
+
+	// Unspecified limit returns both runs.
+	runs, _, err = sys.TaskService.FindRuns(sys.Ctx, influxdb.RunFilter{Task: task.ID})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(runs) != 2 {
+		t.Fatalf("expected 2 runs, got %v", runs)
+	}
+	if runs[0].ID != rc0.Created.RunID {
+		t.Fatalf("retrieved wrong run ID; want %s, got %s", rc0.Created.RunID, runs[0].ID)
+	}
+	if exp := startedAt.Format(time.RFC3339Nano); runs[0].StartedAt != exp {
+		t.Fatalf("unexpectedStartedAt; want %s, got %s", exp, runs[0].StartedAt)
+	}
+	if runs[0].Status != backend.RunStarted.String() {
+		t.Fatalf("unexpected run status; want %s, got %s", backend.RunStarted.String(), runs[0].Status)
+	}
+	if runs[0].FinishedAt != "" {
+		t.Fatalf("expected empty FinishedAt, got %q", runs[0].FinishedAt)
+	}
+
+	if runs[1].ID != rc1.Created.RunID {
+		t.Fatalf("retrieved wrong run ID; want %s, got %s", rc1.Created.RunID, runs[1].ID)
+	}
+	if runs[1].StartedAt != runs[0].StartedAt {
+		t.Fatalf("unexpected StartedAt; want %s, got %s", runs[0].StartedAt, runs[1].StartedAt)
+	}
+	if runs[1].Status != backend.RunSuccess.String() {
+		t.Fatalf("unexpected run status; want %s, got %s", backend.RunSuccess.String(), runs[0].Status)
+	}
+	if exp := startedAt.Add(time.Second).Format(time.RFC3339Nano); runs[1].FinishedAt != exp {
+		t.Fatalf("unexpected FinishedAt; want %s, got %s", exp, runs[1].FinishedAt)
+	}
+
+	// Look for a run that doesn't exist.
+	_, err = sys.TaskService.FindRunByID(sys.Ctx, task.ID, influxdb.ID(math.MaxUint64))
+	if err == nil {
```

Comment:
Should this be `err != backend.ErrRunNotFound`?

Reply:
OO good point. It will have to be `err != backend.ErrRunNotFound && err != kv.ErrRunNotFound`. I will get that in. thanks!